include/fcl/data_types.h

```diff
@@ -45,10 +45,10 @@ namespace fcl
 {
 
 typedef double FCL_REAL;
-typedef std::uint_fast64_t FCL_INT64;
-typedef std::int_fast64_t FCL_UINT64;
-typedef std::uint_fast32_t FCL_UINT32;
-typedef std::int_fast32_t FCL_INT32;
+typedef std::int64_t  FCL_INT64;
+typedef std::uint64_t FCL_UINT64;
+typedef std::int32_t  FCL_INT32;
+typedef std::uint32_t FCL_UINT32;
```

Comment:
> std::int64_t et al. are an optional part of the C++11 standard ...

They are only optional on platforms that can't provide an exactly 32 & 64 bit 2's complement type. According to C++11 7.20.1.1/3 (quoting from stackoverflow):
> However, if an implementation provides integer types with widths of 8, 16, 32, or 64 bits, no padding bits, and (for the signed types) that have a two’s complement representation, it **shall define** the corresponding typedef names.

A more general solution would be to use the "least" types like `std::int_least32_t` which will be exactly the specified if it is available, otherwise the next larger size. The problem with that IMO is that to me the 32 in `FCL_INT32` implies exactly 32 bits by analogy with `int32_t`. I would rename it `FCL_INT_LEAST32` if that was what it means. Actually I would prefer just to remove these macros altogether and use the standard typedefs instead (in another PR).

> What problems occur if there are too many bits?

That would depend on the usage. I believe in the Morton case, the 32 & 64 bit specializations were intended to provide compact storage, so choosing a 64 bit representation for the 32 would be odd (even if that were the fastest type). I would just like the integer size dependencies to be clear. Currently this is kind of a mess in FCL and I'm wondering whether that may account for some of the failures I'm seeing on Windows.

Here are some related discussions: [Google style guide](https://google.github.io/styleguide/cppguide.html#Integer_Types), [Khronos issue](https://github.com/KhronosGroup/Vulkan-Docs/issues/122)